### PR TITLE
[ODPR-1472] Pass contactDetails when registering without ID

### PR DESCRIPTION
**Pull request by paddynski-opencast**
### Description

Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[ODPR-1472](https://jira.tools.tax.service.gov.uk/browse/ODPR-1472)

##### Before creating PR

- [x] Have you run the tests?
- [x] Have you run the journey tests? (where applicable)
- [x] Have you addressed warnings where appropriate?
- [x] Have you rebased against the current version of main?
- [ ] Have you checked code coverage isn’t lower than previously?
- [x] Have you checked to ensure all dependencies are up-to-date?

##### After PRs been raised

- [ ] Have you checked the PR Builder passes?

##### Before Reviewing

- [x] Have you pulled the branch down?
- [x] Have you assigned yourself to the PR?
- [x] Have you moved the task to “in review” on JIRA?
- [x] Have you checked to ensure all dependencies are up-to-date?
- [x] Have you checked to ensure branch has been rebased against the current version of main?

##### Whilst Reviewing

- [x] Have you run the tests?
- [x] Have you run the journey tests?
- [x] Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing

- [ ] Have you checked for merge conflicts or any changes in the current main that may affect the current pull request?
  i.e. does it need another rebase?
- [ ] Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ] Have you moved the task to “in pipeline” on Jira?

### CODE REVIEW

File: app/models/registration/requests/IndividualWithoutId.scala [hunk 1]
```diff
@@ -42,9 +42,18 @@ object IndividualWithoutId {
...
+      answers.getEither(IndividualEmailAddressPage)
+    ).parMapN { (name, dateOfBirth, address, email) =>
+      IndividualWithoutId(
+        name.firstName,
+        name.lastName,
+        dateOfBirth,
+        address,
+        ContactDetails(
+          email,
+          answers.get(IndividualPhoneNumberPage)
```
Comment (georgievh) [inline app/models/registration/requests/IndividualWithoutId.scala:55]: This should not be there, it should be in the same sections where the other similar operations are done.

File: app/models/registration/requests/OrganisationWithoutId.scala [hunk 2]
```diff
@@ -33,8 +37,19 @@ object OrganisationWithoutId {
...
-      answers.getEither(BusinessAddressPage)
-    ).parMapN { (name, businessAddress) =>
-      OrganisationWithoutId(name, Address(businessAddress))
+      answers.getEither(BusinessAddressPage),
+      answers.getEither(PrimaryContactNamePage),
+      answers.getEither(PrimaryContactEmailAddressPage)
+    ).parMapN { (name, businessAddress, contactName, email) =>
+      val contact = ContactDetails(
+        email,
+        answers.get(PrimaryContactPhoneNumberPage)
```
Comment (georgievh) [inline app/models/registration/requests/OrganisationWithoutId.scala:46]: Same here

File: test-utils/builders/ContactBuilder.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package builders
+
+import models.registration.requests.ContactDetails
+
+object ContactBuilder {
+
+  val aContact: ContactDetails =
```
Comment (georgievh) [inline test-utils/builders/ContactBuilder.scala:23]: This should be aContactDetails. Please copy the builder as I have updated it in the BE

File: test/controllers/CheckYourAnswersControllerSpec.scala [hunk 4]
```diff
@@ -310,7 +311,7 @@ class CheckYourAnswersControllerSpec extends SpecBase with MockitoSugar with Bef
                 .set(CanPhoneIndividualPage, false).success.value
                 .set(SoleTraderNamePage, SoleTraderName("first", "last")).success.value
 
-              val expectedRegistrationRequest = IndividualWithoutId("first", "last", aDateOfBirth, Address.fromUkAddress(aUkAddress))
+              val expectedRegistrationRequest = IndividualWithoutId("first", "last", aDateOfBirth, Address.fromUkAddress(aUkAddress), aContact)
```
Comment (georgievh) [inline test/controllers/CheckYourAnswersControllerSpec.scala:314]: Make the data obvious, check the changes I did in your BE commit.

File: test/controllers/CheckYourAnswersControllerSpec.scala [hunk 5]
```diff
@@ -358,7 +359,9 @@ class CheckYourAnswersControllerSpec extends SpecBase with MockitoSugar with Bef
               .set(DateOfBirthPage, aDateOfBirth).success.value
               .set(AddressInUkPage, true).success.value
               .set(UkAddressPage, aUkAddress).success.value
-            val expectedRegistrationRequest = IndividualWithoutId("first", "last", aDateOfBirth, Address.fromUkAddress(aUkAddress))
+              .set(IndividualEmailAddressPage, aContact.emailAddress).success.value
+
+            val expectedRegistrationRequest = IndividualWithoutId("first", "last", aDateOfBirth, Address.fromUkAddress(aUkAddress), aContact)
```
Comment (georgievh) [inline test/controllers/CheckYourAnswersControllerSpec.scala:364]: Same as above and please do the same for all the tests.

**Review by georgievh — CHANGES_REQUESTED** [review]

File: test-utils/builders/ContactBuilder.scala (hunk 3 above)
Comment (paddynski-opencast) [reply test-utils/builders/ContactBuilder.scala]: Done.

File: test/controllers/CheckYourAnswersControllerSpec.scala (hunk 4 above)
Comment (paddynski-opencast) [reply test/controllers/CheckYourAnswersControllerSpec.scala]: Done.

File: test/controllers/CheckYourAnswersControllerSpec.scala (hunk 5 above)
Comment (paddynski-opencast) [reply test/controllers/CheckYourAnswersControllerSpec.scala]: Done.

File: app/models/registration/requests/IndividualWithoutId.scala (hunk 1 above)
Comment (paddynski-opencast) [reply app/models/registration/requests/IndividualWithoutId.scala]: Moved.

**Review by Rabina-Rai — APPROVED** [review]

**Review by georgievh — APPROVED** [review]

**Pull request merged**